src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1118,19 +1128,132 @@ map_injective f.injective u v
 
 end path
 
+/-! ## Transferring between graphs -/
+
+namespace walk
+
+variables {G}
+
+/-- The walk `p` transferred to lie in `H` -/
+@[simp] def transfer : Π {u v : V} (p : G.walk u v) (H : simple_graph V)
+  (h : ∀ e, e ∈ p.edges → e ∈ H.edge_set), H.walk u v
+| _ _ (walk.nil) H h := walk.nil
+| _ _ (walk.cons a p) H h := by
+  { refine walk.cons _ (p.transfer H _);
+    simp only [walk.edges_cons, list.mem_cons_iff, forall_eq_or_imp, mem_edge_set] at h,
+    exact h.1, exact h.2, }
+
+variables {u v w : V} (p : G.walk u v) (q : G.walk v w)
+  {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set)
+  (hq : ∀ e, e ∈ q.edges → e ∈ H.edge_set)
+
+lemma transfer_id : p.transfer G (λ e ep, edges_subset_edge_set p ep) = p := by
+begin
+  induction p,
+  { simp only [transfer] },
+  { simp only [p_ih, transfer, eq_self_iff_true, heq_iff_eq, and_self] },
+end
+
+/-- The walk `p` transfered to the larger graph `H` -/
+abbreviation transfer_le (GH : G ≤ H) : H.walk u v :=
+p.transfer H (λ e ep, edge_set_mono GH (edges_subset_edge_set p ep))
+
+lemma transfer_eq_map_spanning_subgraphs (GH : G ≤ H) :
+  p.transfer H hp = p.map (simple_graph.hom.map_spanning_subgraphs GH) :=
+begin
+  induction p,
+  { simp only [transfer, map_nil], },
+  { simp only [p_ih, transfer, map_cons, hom.map_spanning_subgraphs_apply, eq_self_iff_true,
+             heq_iff_eq, and_self], },
+end
+
+@[simp] lemma edges_transfer : (p.transfer H hp).edges = p.edges :=
+begin
+  induction p,
+  { simp only [transfer, edges_nil], },
+  { simp only [p_ih, transfer, edges_cons, eq_self_iff_true, and_self], },
+end
+
+@[simp] lemma support_transfer : (p.transfer H hp).support = p.support :=
+begin
+  induction p,
+  { simp only [transfer, support_nil, eq_self_iff_true, and_self], },
+  { simp only [p_ih, transfer, support_cons, eq_self_iff_true, and_self], },
+end
+
+lemma is_path_transfer (pp : p.is_path) : (p.transfer H hp).is_path :=
+begin
+  induction p,
+  { simp only [transfer, is_path.nil], },
+  { simp only [cons_is_path_iff, transfer, support_transfer] at pp ⊢,
+  simp only [p_ih, pp, not_false_iff, and_self], },
+end
+
+lemma is_cycle_transfer {u : V} (p : G.walk u u) {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set) (pc : p.is_cycle) : (p.transfer H hp).is_cycle :=
+begin
+  cases p,
+  { simp only [transfer, is_cycle.not_of_nil] at pc ⊢, exact pc, },
+  { simp only [path.cons_is_cycle_iff, transfer, edges_transfer] at pc ⊢,
+    refine ⟨_, pc.right⟩,
+    apply is_path_transfer,
+    exact pc.left, },
+end
+
+lemma is_cycle_transfer_le {u : V} (p : G.walk u u) {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set) (pc : p.is_cycle)
```

Comment:
```suggestion
lemma is_cycle_transfer_le {u : V} (p : G.walk u u) {H : simple_graph V} (hp) (pc : p.is_cycle)
```
and you could get rid of one more line if you were to move `{u : V} {H : simple_graph V}` to a `variables` declaration.

Reply:
Thanks, I thought that since `p` is not of the same type as defined in the `variables …` part above, I would have to provide `hp`'s type explicitely again.

Reply:
In Lean, when you write `(hp)` as a variable it's short for `(hp : _)` and then Lean tries to figure out the `_`. This is one of these things you can decide to do on a case-by-case basis. I tend to like to go for writing out types, but sometimes in context this style is clear enough.

Reply:
While every line of code has a maintenance cost, sometimes these are lines well spent since they provide context and make maintenance less expensive.

I unresolved this conversation so you can see it, but please resolve it again once you have.